src/drivers/barometer/bmp581/bmp581.h

```diff
@@ -0,0 +1,345 @@
+#define BMP5_IIR_FILTER_COEFF_1		(0x01)
+#define BMP5_IIR_FILTER_COEFF_3		(0x02)
+#define BMP5_IIR_FILTER_COEFF_7		(0x03)
+#define BMP5_IIR_FILTER_COEFF_15	(0x04)
+#define BMP5_IIR_FILTER_COEFF_31	(0x05)
+#define BMP5_IIR_FILTER_COEFF_63	(0x06)
+#define BMP5_IIR_FILTER_COEFF_127	(0x07)
+
+/* Macro is used to bypass both iir_t and iir_p together */
+#define BMP5_IIR_BYPASS                 (0xC0)
+
+/* Interrupt configurations */
+#define BMP5_INT_MODE_PULSED            (0)
+#define BMP5_INT_MODE_LATCHED           (1)
+
+#define BMP5_INT_POL_ACTIVE_LOW         (0)
+#define BMP5_INT_POL_ACTIVE_HIGH        (1)
+
+#define BMP5_INT_OD_PUSHPULL            (0)
+#define BMP5_INT_OD_OPENDRAIN           (1)
+
+/* NVM and Interrupt status asserted macros */
+#define BMP5_INT_ASSERTED_POR_SOFTRESET_COMPLETE  (0x10)
+#define BMP5_INT_NVM_RDY                          (0x02)
+#define BMP5_INT_NVM_ERR                          (0x04)
+#define BMP5_INT_NVM_CMD_ERR                      (0x08)
+
+/* Interrupt configurations */
+#define BMP5_INT_MODE_MSK               (0x01)
+
+#define BMP5_INT_POL_MSK                (0x02)
+#define BMP5_INT_POL_POS                (1)
+
+#define BMP5_INT_OD_MSK                 (0x04)
+#define BMP5_INT_OD_POS                 (2)
+
+#define BMP5_INT_EN_MSK                 (0x08)
+#define BMP5_INT_EN_POS                 (3)
+
+#define BMP5_INT_DRDY_EN_MSK            (0x01)
+
+#define BMP5_INT_FIFO_FULL_EN_MSK       (0x02)
+#define BMP5_INT_FIFO_FULL_EN_POS       (1)
+
+#define BMP5_INT_FIFO_THRES_EN_MSK      (0x04)
+#define BMP5_INT_FIFO_THRES_EN_POS      (2)
+
+#define BMP5_INT_OOR_PRESS_EN_MSK       (0x08)
+#define BMP5_INT_OOR_PRESS_EN_POS       (3)
+
+/* ODR configuration */
+#define BMP5_ODR_MSK                    (0x7C)
+#define BMP5_ODR_POS                    (2)
+
+/* OSR configurations */
+#define BMP5_TEMP_OS_MSK                (0x07)
+
+#define BMP5_PRESS_OS_MSK               (0x38)
+#define BMP5_PRESS_OS_POS               (3)
+
+/* Pressure enable */
+#define BMP5_PRESS_EN_MSK               (0x40)
+#define BMP5_PRESS_EN_POS               (6)
+
+/* IIR configurations */
+#define BMP5_SET_IIR_TEMP_MSK           (0x07)
+
+#define BMP5_SET_IIR_PRESS_MSK          (0x38)
+#define BMP5_SET_IIR_PRESS_POS          (3)
+
+#define BMP5_SHDW_SET_IIR_TEMP_MSK      (0x08)
+#define BMP5_SHDW_SET_IIR_TEMP_POS      (3)
+
+#define BMP5_SHDW_SET_IIR_PRESS_MSK     (0x20)
+#define BMP5_SHDW_SET_IIR_PRESS_POS     (5)
+
+#define BMP5_IIR_FLUSH_FORCED_EN_MSK    (0x04)
+#define BMP5_IIR_FLUSH_FORCED_EN_POS    (2)
+
+/* Powermode */
+#define BMP5_POWERMODE_MSK		(0x03)
+
+#define BMP5_DEEP_DISABLE_MSK		(0x80)
+#define BMP5_DEEP_DISABLE_POS		(7)
+
+/* Fifo configurations */
+#define BMP5_FIFO_FRAME_SEL_MSK        (0x03)
+
+/* NVM and Interrupt status asserted macros */
+#define BMP5_INT_ASSERTED_DRDY         (0x01)
+
+/*!
+ * @brief Enumerator for powermode selection
+ */
+enum bmp5_powermode {
+    /*! Standby powermode */
```

Comment:
these comments are redundant, the enum names already describe themselves. Remove redundant comments throughout